pkg/render/applicationlayer/applicationlayer.go

```diff
@@ -149,20 +155,28 @@ func (c *component) Objects() ([]client.Object, []client.Object) {
 	// If l7spec is provided render the required objects.
 	objs = append(objs, c.serviceAccount())
 
-	c.config.dikastesEnabled = false
-	if c.config.WAFEnabled || c.config.ALPEnabled {
-		c.config.dikastesEnabled = true
-	}
+	c.config.dikastesEnabled = c.config.WAFEnabled ||
+		c.config.ALPEnabled ||
+		c.config.SidecarInjectionEnabled
+
+	c.config.l7logcollectorEnabled = c.config.LogsEnabled ||
+		c.config.SidecarInjectionEnabled
 
-	// If Web Application Firewall is enabled, we need WAF ruleset ConfigMap present.
-	if c.config.WAFEnabled {
+	c.config.envoyEnabled = c.config.WAFEnabled ||
+		c.config.ALPEnabled ||
+		c.config.LogsEnabled
+
+	// If Web Application Firewall or Sidecar Injection is enabled, we need WAF ruleset ConfigMap present.
+	if c.config.WAFEnabled || c.config.SidecarInjectionEnabled {
 		// this ConfigMap is a copy of the provided configuration from the operator namespace into the calico-system namespace
 		objs = append(objs, c.modSecurityConfigMap())
 	}
 
 	// Envoy configuration
-	c.config.envoyConfigMap = c.envoyL7ConfigMap()
-	objs = append(objs, c.config.envoyConfigMap)
+	if c.config.envoyEnabled {
+		c.config.envoyConfigMap = c.envoyL7ConfigMap()
+		objs = append(objs, c.config.envoyConfigMap)
+	}
 
 	// Envoy & Dikastes Daemonset
 	objs = append(objs, c.daemonset())
```

Comment:
Do we need to install the daemonset even if we're only configured to use sidecars (and not per-host envoy)?

If so, we should have a detailed comment explaining why.